### DISCUSSION TITLE
P3018: Help me understand how to deploy my schema

### DISCUSSION THREAD

**Discussion by mahmoudalsof** [prisma/prisma#7116] [Q&A] [answered]
My current workflow (or at least what I think is correct) is do my changes while developing using `prisma migrate dev --name [migration-name]` and then merge it to the production branch. 

Then, on the server I would pull from the production branch and verify that the newly created `migration` folder has been created. 

But when I run `prisma migrate deploy` I get the following error every single time I've tried.

```
Error: P3018

A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20210516163425_change_comment_to_text

Database error code: 1146

Database error:
Table 'dbName.tableName' doesn't exist
```

and I have no idea what I'm doing wrong. I have existing data in the database that I do not want to lose. Is my understanding of migrations correct or is there something I'm missing here?

**Comment by mahmoudalsof** [comment, marked as answer]
So everything I did was correct. I found out the the `.sql` file generated in the migrations folder has the table name as lowercase where in production, my table names are in pascal case. Is there a way to set this with the CLI ?

**Reply by ryands17** [reply]
@MahmoudAlSofyani 👋 

Before using Migrate, did your database already have tables? If so, you need to follow [this](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project) tutorial to get all your migrations done properly.

**Reply by mahmoudalsof** [reply]
@ryands17 Sorry for the late response. Yes the database already had the initial prisma migration applied. I don't think the link you sent helps me. The issue I have is the generated `.sql` file has the tables in lowercase where in my production database, they are in pascal case. What I find interesting is that in my development enviroment, the resulting database follows the naming convention of the generated prisma migration files. 
I suspect this may just be a mysql configuration i need to customize.

**Reply by mahmoudalsof** [reply]
@ryands17 so another issue presented itself. Apparently my prisma version and my prisma client versions were not identical. This fixed another issue I was facing but I'm not entirely sure if this will fix the lowercase issue.

**Reply by ryands17** [reply]
@MahmoudAlSofyani 👋 

It would be great if you could open an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with the reproduction steps so that we can look into this :)

**Reply by mahmoudalsof** [reply]
@ryands17 Appreciate the help. I have submitted an issue #7250
